docs/gallery/interactive/pyfar_plotting.ipynb

```diff
@@ -0,0 +1,396 @@
+    "One for each subplot."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Spectrograms\n",
+    "\n",
+    "`pyfar` also provides functions to create spectrograms."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pf.plot.spectrogram(sweep)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Similar to the line plots, the spectrogram can be customized using `pyfar` specific arguments as well as `matplotlib` specific arguments.\n",
+    "Note, that either `matplotlib.pyplot.pcolormesh()` or `matplotlib.pyplot.contourf()` is used depending on the selected `method`.\n",
+    "As such, also the available kwargs might change."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pf.plot.spectrogram(sweep,\n",
+    "                    # pyfar arguments\n",
+    "                    window_overlap_fct =0.75,\n",
+    "                    window_length=2**9,\n",
+    "                    unit=\"ms\",\n",
+    "                    # matplotlib arguments\n",
+    "                    cmap=\"viridis\",\n",
+    "                )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2D plots\n",
+    "\n",
+    "There are also functions to plot a multi-channel `pyfar.Signal` in 2D.\n",
+    "All line plots have such a corresponding 2D plot method."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sample_delays = np.rint(8 * np.sin(np.linspace(0, 2*np.pi, 25))).astype(int) + 32\n",
+    "impulses = pf.signals.impulse(64, sample_delays, np.linspace(1, .5, 25))\n",
+    "pf.plot.time_2d(impulses, unit='ms')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Styling plots\n",
+    "\n",
+    "`pyfar` includes two pyfar-specific plot styles: _light_ and _dark_.\n",
+    "The default style is _light_.\n",
+    "\n",
+    "The `pyfar` plot functions accept a `style` argument to change the style."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "signal = pf.Signal([0, 1, 0, -1], 44100)\n",
+    "\n",
+    "pf.plot.time(signal, style=\"dark\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The overall style can be temporarily changed using `pyfar.plot.context` which is a wrapper around `matplotlib.context`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with pf.plot.context(style=\"dark\"):\n",
+    "    fig, ax = plt.subplots(1, 1)\n",
+    "    pf.plot.time(signal, ax=ax)\n",
+    "\n",
+    "plt.figure()\n",
+    "pf.plot.time(signal)"
```

Comment:
Creating a figure this way within the same code block results in only partially applying "light" style, which leads to more differences than just "dark" vs. "light". I'd suggest to remove it.
```suggestion
```